src/app/sidebar/sidebar.component.ts

```diff
@@ -56,6 +56,7 @@ export class SidebarComponent implements OnInit {
     fetchUserData() {
         this.userService.getOwnProfile().subscribe(res => {
             this.fullName = res['first_name'] + ' ' + res['last_name'];
+            this.hasVerifiedProfile = res['verified'];
```

Comment:
Has this been tested that this function is called _after_ the user finishes Identyum verification? Was this achieved by @mdrusko1 with the `refresh` functionality in other pull request maybe? (not sure, just asking)

Reply:
Yes, I'm concerned about that too. 

As the sidebar component stays the same during the entire application session, it should be notified about changing user data from outside. I propose creating a [Subject](https://www.learnrxjs.io/learn-rxjs/subjects/subject) inside user service which will act as a notifier for any other component that needs to change when the user data changes from outside. For example, when a user token needs to be changed, you will get new data and pass it to the subject. Now, inside the sidebar component, inject user service and subscribe to this subject which will tell the sidebar component when to refresh the user data.

Tell me if it's confusing. :)